File: rbscript/client_test.rb
require "net/http"
require "uri"
require "json"

def test_public_api
  # インターネット上のテスト用API（echoサービス）
  url = "https://httpbin.org/get"

  # テストとして、APIに送る適当なクエリパラメータ
  params = { "message" => "Hello bundler!", "status" => "success" }

  puts "〇 インターネット上のAPIに接続中..."

  uri = URI(url)
  uri.query = URI.encode_www_form(params)

  begin
    # GETリクエストを送信
    response = Net::HTTP.get_response(uri)

    # ステータスコードが200（成功）かチェック
    unless response.is_a?(Net::HTTPSuccess)
      raise "HTTP #{response.code} #{response.message}"
    end

    # レスポンスのJSONを解析
    data = JSON.parse(response.body)

    puts "\n✅ 接続成功！サーバーから返ってきたデータ:"
    puts "あなたのIPアドレス: #{data['origin']}"
    puts "送信したパラメータ: #{data['args']}"
    puts "//受信データ: #{data}"
  rescue => e
    puts "❌ 通信エラーが発生しました: #{e.message}"
  end
end

test_public_api if __FILE__ == $PROGRAM_NAME

File: jsscript/client_test.js
import axios from 'axios';

async function testPublicApi() {
    // インターネット上のテスト用API（echoサービス）
    const url = 'https://httpbin.org/get';

    // テストとして、APIに送る適当なクエリパラメータ
    const params = {
        message: 'Hello pnpm!',
        status: 'success'
    };

    console.log('○ インターネット上のAPIに接続中...');

    try {
        // GETリクエストを送信
        const response = await axios.get(url, { params });

        // レスポンスのデータを取得
        const data = response.data;

        console.log('\n✅ 接続成功！サーバーから返ってきたデータ:');
        console.log(`あなたのIPアドレス: ${data.origin}`);
        console.log('送信したパラメータ:', data.args);
        console.log('受信データ:', data);

    } catch (error) {
        console.error(`❌ 通信エラーが発生しました: ${error.message}`);
    }
}

testPublicApi();
